WordPress/openverse-frontend
# Share audio loaded state through pinia

## Fixes


Fixes #1528  by @zackkrida 

## Description


Based on my debugging it seemed like the issue was that the `loaded` event was never firing for the audio after it had already been loaded on the search results page, so when the audio track component is initialized on the single result page, `hasLoaded` is set to false and never updated to `true` because the `loaded` event doesn't fire again after the first time. This made the status get stuck on `loading`.
This PR now follows @obulat's suggestion to manage the audio state within the audio object itself so that it's easily shared throughout the app without the need for an additional store. This treats the audio object itself as the source of truth. It works great I think! And is maybe a little less unexpected. Thanks for the suggestion @obulat!

## Testing Instructions

Follow the repro instructions in the linked issue and confirm it's no longer reproducible.

I'm not sure whether to add an e2e test for this as I don't know whether we can reliably mock audio responses at the moment? It's probably worth looking into though.

## Checklist

- [N/A] I added or updated documentation (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
